src/pages/Main/SearchForm/SearchForm.js

```diff
@@ -1,23 +1,65 @@
 import React from 'react';
+import { withRouter } from 'react-router-dom';
 import './SearchForm.scss';
 
 class SearchForm extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      searchValue: '',
+      matchResult: [],
+    };
+  }
+
+  searchText = e => {
+    this.setState({
+      searchValue: e.target.value,
+    });
+  };
+
+  handleSubmit = e => {
+    e.preventDefault();
+    const { searchValue } = this.state;
+    if (searchValue.length === 0) {
+      alert('검색어를 입력해 주세요.');
+    } else {
+      fetch(`http://10.58.2.176:8000/restaurants/search?search=${searchValue}`)
+        .then(res => res.json())
+        .then(data => {
+          this.setState({
+            matchResult: data,
+          });
+          const { matchResult } = this.state;
+          this.props.history.push({
+            pathname: './ShopList',
+            state: { searchValue: searchValue, matchResult: matchResult },
```

Comment:
```suggestion
            state: { searchValue, matchResult },
```
이름이 동일한 경우에는 이렇게 축약해줄 수도 있습니다.